src/origdatablocks/origdatablocks.controller.ts

```diff
@@ -263,12 +347,26 @@ export class OrigDatablocksController {
   )
   @Get("/fullfacet")
   async fullfacet(
+    @Req() request: Request,
     @Query() filters: { fields?: string; facets?: string },
   ): Promise<Record<string, unknown>[]> {
+    const user: JWTUser = request.user as JWTUser;
+    const fields: IOrigDatablockFields = JSON.parse(filters.fields ?? "{}");
```

Comment:
there are a lot of similar blocks like this,
I guess we should try to refactor it to use a common function. 
something like (pseudo-code):
```
function accessGroupsFilterFields( fields_in, user, OrigDataBlock) {
  if (user) {
   ability = ...
  canViewAll = ...
 if ( !canViewAll ){
   ...
     fields_in.userGroups.push(...)
   }
return fields_in
}
```

Reply:
@bpedersen2 I will postpone code refactoring to a new issue.
I'm going through a review of all the authorization for datasets and connected entities, which will require some changes.
We chat tomorrow during the meeting if you are present.

Reply:
I've created new issue to address this issue #821